addon/utils/game.py

```diff
@@ -1,19 +1,38 @@
 from pathlib import Path
 from ..utils.common import resolve
-
+import os
 
 def update_game(self, context):
-    self['game'] = resolve(self.game)
+    g = resolve(self.game)
+    # Don't bother re-resolving paths if we haven't changed our game path
+    if g == self['game']:
+        return
+
+    self['game'] = g
     game = Path(self.game)
 
     if game.joinpath('gameinfo.txt').is_file():
         bin = game.parent.joinpath('bin')
 
-        if not bin.joinpath('studiomdl.exe').is_file():
-            for path in bin.iterdir():
-                if path.is_dir() and path.joinpath('studiomdl.exe').is_file():
-                    bin = path
-                    break
+        # If we're not on the old-style pattern bin/<something> layout, check for platform subdirs
+        actualbin = None
+        if not bin.joinpath('studiomdl.exe').is_file() and not bin.joinpath('studiomdl').is_file():
+            def check_subdir(subdirs, smdl):
+                for s in subdirs:
+                    p = bin.joinpath(s)
+                    if p.is_dir() and p.joinpath(smdl).is_file():
+                        return p
+                return None
+
+            # For linux, prefer the native binaries (if possible)
+            if os.name == 'posix':
+                actualbin = check_subdir(['linux32', 'linux64'], 'studiomdl')
+            # Resolve windows paths
+            if actualbin is None:
+                actualbin = check_subdir(['win32', 'win64'], 'studiomdl.exe')
+
+        if actualbin is not None:
+            bin = actualbin
```

Comment:
Kind of a nitpick but I'm not a fan of these short variable names, I think renaming s to subdir and p to path would make it easier to read.
I'll probably refactor this function a little anyway after merging though.